components/tikv_util/src/mpsc/batch.rs

```diff
@@ -298,6 +298,69 @@ impl<T, E, I: Fn() -> E, C: Fn(&mut E, T)> Stream for BatchReceiver<T, E, I, C>
     }
 }
 
+/// `SizedBatchReceiver` is a `futures::Stream`, which returns a batched type.
+/// It receives the size of the sent item, uses this size to compute the batch size.
+pub struct SizedBatchReceiver<T, E, I: Fn() -> E, C: Fn(&mut E, T)> {
```

Comment:
Can it be replaced by [`Stream::chunks`](https://docs.rs/futures/0.1.29/futures/stream/trait.Stream.html#method.chunks)?

Reply:
Have replaced it with `flattenMap`.